docs/kernel.rst

```diff
@@ -1,4 +1,243 @@
+.. _kernel:
+
 Kernel
 ======
 
-contains Kernel details
\ No newline at end of file
+A kernel models a circuit with quantum gates ending in one or more measurements.
+In OpenQL, this has been extended with:
+
+- control flow that can jump at the end of a kernel to the start of another kernel; see this section
+- classical gates mixed with quantum gates in a single circuit with the design objective to support control flow changes; see :ref:`classical_instructions`
+
+In OpenQL a ``kernel`` is an object; it has a name, a type,
+and a circuit as its main structural attributes.
```

Comment:
Yes, that is true. That is because control flow cannot be scheduled now. If it could, we could schedule the whole program, with gates having cycles relative to the start of each kernel. There is a note on this in the section control_flow_in_the_internal_representation